apps/meteor/client/views/admin/moderation/UserReports/ModConsoleUsersTable.tsx

```diff
@@ -0,0 +1,148 @@
+import { Pagination, States, StatesAction, StatesActions, StatesIcon, StatesTitle } from '@rocket.chat/fuselage';
+import { useDebouncedValue, useMediaQuery, useMutableCallback } from '@rocket.chat/fuselage-hooks';
+import { useEndpoint, useTranslation, useRouter } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+import type { FC } from 'react';
+import React, { useMemo, useState } from 'react';
+
+import GenericNoResults from '../../../../components/GenericNoResults';
+import {
+	GenericTable,
+	GenericTableLoadingTable,
+	GenericTableHeaderCell,
+	GenericTableBody,
+	GenericTableHeader,
+} from '../../../../components/GenericTable';
+import { usePagination } from '../../../../components/GenericTable/hooks/usePagination';
+import { useSort } from '../../../../components/GenericTable/hooks/useSort';
+import ModerationFilter from '../helpers/ModerationFilter';
+import ModConsoleUserTableRow from './ModConsoleUserTableRow';
+
+const ModConsoleUsersTable: FC = () => {
+	const [text, setText] = useState('');
+	const router = useRouter();
+	const t = useTranslation();
+	const isDesktopOrLarger = useMediaQuery('(min-width: 1024px)');
+
+	const { sortBy, sortDirection, setSort } = useSort<
+		'reports.ts' | 'reports.reportedUser.username' | 'reports.reportedUser.createdAt' | 'count'
+	>('reports.ts');
+	const { current, itemsPerPage, setItemsPerPage: onSetItemsPerPage, setCurrent: onSetCurrent, ...paginationProps } = usePagination();
+
+	const [dateRange, setDateRange] = useState<{ start: string | null; end: string | null }>({
+		start: '',
+		end: '',
+	});
+	const { start, end } = dateRange;
+
+	const debouncedText = useDebouncedValue(text, 500);
+
+	const query = useMemo(
+		() => ({
+			selector: debouncedText,
+			sort: JSON.stringify({ [sortBy]: sortDirection === 'asc' ? 1 : -1 }),
+			count: itemsPerPage,
+			offset: current,
+			latest: end ? `${new Date(end).toISOString().slice(0, 10)}T23:59:59.999Z` : undefined,
+			oldest: start ? `${new Date(start).toISOString().slice(0, 10)}T00:00:00.000Z` : undefined,
+		}),
+		[current, debouncedText, end, itemsPerPage, sortBy, sortDirection, start],
+	);
```

Comment:
there is no need to memoize this object , usequery always serialize the object resulting in a hash

Reply:
Nice catch. Thank you! I have corrected it.